pkgs/development/guile-modules/guile-gnome/default.nix

```diff
@@ -1,58 +1,41 @@
 { fetchurl, stdenv, guile, guile-lib, gwrap
 , pkgconfig, gconf, glib, gnome_vfs, gtk2
 , libglade, libgnome, libgnomecanvas, libgnomeui
-, pango, guile-cairo, autoconf, automake, texinfo }:
+, pango, guile-cairo, autoconf, automake, texinfo
+}:
 
 stdenv.mkDerivation rec {
-  name = "guile-gnome-platform-2.16.4";
+  name = "${pname}-${version}";
+  pname = "guile-gnome-platform";
+  version = "2.16.4";
 
   src = fetchurl {
-    url = "http://ftp.gnu.org/pub/gnu/guile-gnome/guile-gnome-platform/${name}.tar.gz";
+    url = "http://ftp.gnu.org/pub/gnu/guile-gnome/${pname}/${name}.tar.gz";
     sha256 = "adabd48ed5993d8528fd604e0aa0d96ad81a61d06da6cdd68323572ad6c216c3";
   };
 
   buildInputs = [
-    autoconf
-    automake
-    texinfo
-    guile
-    gwrap
-    pkgconfig
-    gconf
-    glib
-    gnome_vfs
-    gtk2
-    libglade
-    libgnome
-    libgnomecanvas
-    libgnomeui
-    pango
-    guile-cairo
+    autoconf automake texinfo guile gwrap pkgconfig gconf glib gnome_vfs gtk2
+    libglade libgnome libgnomecanvas libgnomeui pango guile-cairo
   ] ++ stdenv.lib.optional doCheck guile-lib;
 
-  preConfigure = ''
-      ./autogen.sh
-  '';
+  preConfigure = "./autogen.sh";
```

Comment:
1. Multiline strings are preferred for scripts, even for single line ones.
2. `autoreconfHook` is generally preferred to `./autogen.sh`.
3. This package builds even without autoreconf.

Could you also add `GUILE_AUTO_COMPILE = 0;` to silence warnings?